src/draw/DefaultDrawer.js

```diff
@@ -503,7 +546,12 @@ class DefaultDrawer {
     const treemapLayout = d3.treemap()
       .size([this.width, this.height])
       .tile((data) => {
-        const lines = this.__buildLines(data);
+        const newComponents = data.children.filter((child) => !child.data.drawOption);
+        const existingComponents = data.children.filter((child) => child.data.drawOption);
```

Comment:
why don't you:

```js
const components = data.children.map((child) => {
  if (!child.data.drawObtion) {
    this.initializeComponentDrawOptions(child); // to check or something else
  }
  return child;
});

Reply:
Because, in this case, ordering is also important. To minimize overlap, new components should only be placed after existing ones have been placed.